tests/e2e/test_diffusion.py

```diff
@@ -1106,6 +1138,68 @@ def test_diffusion_lora(
     ensure_similar_images(predicted_image, expected_image, min_psnr=35, min_ssim=0.98)
 
 
+@no_grad()
+def test_diffusion_sdxl_batch2(sdxl_ddim: StableDiffusion_XL) -> None:
+    sdxl = sdxl_ddim
+
+    prompt1 = "professional portrait photo of a girl, photograph, highly detailed face, depth of field, moody light, golden hour, style by Dan Winters, Russell James, Steve McCurry, centered, extremely detailed, Nikon D850, award winning photography"
+    negative_prompt1 = "3d render, cartoon, drawing, art, low light, blur, pixelated, low resolution, black and white"
+    prompt2 = "professional portrait photo of a boy"
+    negative_prompt2 = "black and white"
+
+    clip_text_embedding_b2, pooled_text_embedding_b2 = sdxl.compute_clip_text_embedding(
+        text=[prompt1, prompt2], negative_text=[negative_prompt1, negative_prompt2]
+    )
+
+    time_ids = sdxl.default_time_ids
+    time_ids_b2 = sdxl.default_time_ids.repeat(2, 1)
+    sdxl.set_inference_steps(40)
+
+    manual_seed(seed=2)
+    x_b2 = torch.randn(2, 4, 128, 128, device=sdxl.device, dtype=sdxl.dtype)
+    x_1 = x_b2[0:1]
+    x_2 = x_b2[1:2]
+
+    x_b2 = sdxl(
+        x_b2,
+        step=sdxl.steps[0],
+        clip_text_embedding=clip_text_embedding_b2,
+        pooled_text_embedding=pooled_text_embedding_b2,
+        time_ids=time_ids_b2,
+    )
+    predicted_image_b2 = sdxl.lda.latents_to_images(x_b2)
+
+    clip_text_embedding_1, pooled_text_embedding_1 = sdxl.compute_clip_text_embedding(
+        text=prompt1, negative_text=negative_prompt1
+    )
+
+    x_1 = sdxl(
+        x_1,
+        step=sdxl.steps[0],
+        clip_text_embedding=clip_text_embedding_1,
+        pooled_text_embedding=pooled_text_embedding_1,
+        time_ids=time_ids,
+    )
+    predicted_image_1 = sdxl.lda.latents_to_image(x_1)
+
+    clip_text_embedding_2, pooled_text_embedding_2 = sdxl.compute_clip_text_embedding(
+        text=prompt2, negative_text=negative_prompt2
+    )
+
+    x_2 = sdxl(
+        x_2,
+        step=sdxl.steps[0],
+        clip_text_embedding=clip_text_embedding_2,
+        pooled_text_embedding=pooled_text_embedding_2,
+        time_ids=time_ids,
+    )
+
+    predicted_image_2 = sdxl.lda.latents_to_image(x_2)
+
+    ensure_similar_images(predicted_image_b2[0], predicted_image_1, min_psnr=35, min_ssim=0.98)
+    ensure_similar_images(predicted_image_b2[1], predicted_image_2, min_psnr=35, min_ssim=0.98)
```

Comment:
I was thinking that, perhaps, given you only perform one diffusion step, it'd be simpler just to compare the noise predictions (torch allclose) and not VAE decode. WDYT?

Reply:
It would makes sense to compare latents directly.

Not sure about comparing noise.
What we want to test here is e2e "batch-stability" and then including the noise -> latents solver/cfg logic in the test makes sense.

WDYT ?

Reply:
FYI, switching to `torch.allclose` makes the test fail

```python
torch.norm(x_b2[1] - x_2[0])
=> 
tensor(1.4638, device='cuda:0')
```

Studying this, it might take some time

Reply:
1.46 was a bug, but still need to put absolute tolerance to 5e-3 (which seems quite high) see https://github.com/finegrain-ai/refiners/pull/263#issuecomment-1956404911 for the details